### PR TITLE
[IMP] mail, *: convert ThreadPreviewView to use Channel instead of Thread

### PR DESCRIPTION
**Pull request by oomsveta**
\* = im_livechat, rating, test_mail_full

Enterprise: https://github.com/odoo/enterprise/pull/30921

Part of task-2948676.

### CODE REVIEW

**Review by seb-odoo — APPROVED** [review]
I'm tired but the diff looks ok.

File: addons/mail/static/src/models/notification_list_view.js [hunk 1]
```diff
@@ -13,6 +13,32 @@ registerModel({
...
+                    if (t1.lastMessage) {
+                        return -1;
+                    }
+                    if (t2.lastMessage) {
+                        return 1;
+                    }
+                    return t1.id < t2.id ? -1 : 1;
+                })
+                .map(thread => ({ thread }));
+        },
```
Comment (alexkuhn) [inline addons/mail/static/src/models/notification_list_view.js:41]: to do in future commits: turn this into a proper sorted field

File: addons/mail/static/src/models/notification_list_view.js [hunk 2]
```diff
@@ -54,32 +54,30 @@ registerModel({
...
                         )
                         .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
                 }
                 case 'all': {
                     // "All" filter is for channels and chats
-                    return this.messaging.models['Thread']
-                        .all(thread => thread.isPinned && thread.model === 'mail.channel')
+                    return this.messaging.models['Channel']
+                        .all(channel => channel.thread.isPinned)
                         .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
```
Comment (alexkuhn) [inline addons/mail/static/src/models/notification_list_view.js:81]: again to do in future PR: properly define `sort` on this field

**Review by alexkuhn — COMMENTED** [review]
LGTM

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ rebase-ff

**Pull request closed without merging**